api/lcp/server.py

```diff
@@ -124,55 +142,39 @@ class LCPServer:
 
     def __init__(
         self,
-        configuration_storage,
-        configuration_factory,
+        get_configuration,
```

Comment:
Rather than a configuration factory and storage we use a get_configuration Callable to ensure the configuration values we pull at runtime are always pulled from the database and not an in-memory object.